packages/superset-ui-legacy-preset-chart-big-number/src/BigNumber/BigNumber.tsx

```diff
@@ -0,0 +1,309 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import shortid from 'shortid';
+import { OverlayTrigger, Tooltip } from 'react-bootstrap';
+import { t } from '@superset-ui/translation';
+import { getNumberFormatter } from '@superset-ui/number-format';
+import { XYChart, AreaSeries, CrossHair, LinearGradient } from '@data-ui/xy-chart';
+import { BRAND_COLOR } from '@superset-ui/color';
+import { computeMaxFontSize } from '@superset-ui/dimension';
+import NumberFormatter from '@superset-ui/number-format/lib/NumberFormatter';
+import { smartDateVerboseFormatter } from '@superset-ui/time-format';
+import TimeFormatter from '@superset-ui/time-format/lib/TimeFormatter';
+
+import './BigNumber.css';
+
+const defaultNumberFormatter = getNumberFormatter(undefined);
+
+const CHART_MARGIN = {
+  top: 4,
+  right: 4,
+  bottom: 4,
+  left: 4,
+};
+
+const PROPORTION = {
+  // text size: proportion of the chart container sans trendline
+  HEADER: 0.3,
+  SUBHEADER: 0.125,
+  // trendline size: proportion of the whole chart container
+  TRENDLINE: 0.3,
+};
+
+type TimeSeriesDatum = {
+  x: number; // timestamp as a number
+  y: number | null;
+};
+
+export function renderTooltipFactory(
+  formatDate = smartDateVerboseFormatter,
+  formatValue = defaultNumberFormatter,
+) {
+  return function renderTooltip({ datum: { x, y } }: { datum: TimeSeriesDatum }) {
+    // even though `formatDate` supports timestamp as numbers, we need
+    // `new Date` to pass type check
+    return (
+      <div style={{ padding: '4px 8px' }}>
+        {formatDate(new Date(x))}
+        <br />
+        <strong>{y === null ? t('N/A') : formatValue(y)}</strong>
+      </div>
+    );
+  };
+}
+
+type BigNumberVisProps = {
+  className?: string;
+  width: number;
+  height: number;
+  bigNumber: number;
+  bigNumberFallback?: TimeSeriesDatum;
+  formatNumber: NumberFormatter;
+  formatTime: TimeFormatter;
+  fromDatetime: number;
+  toDatetime: number;
+  headerFontSize: number;
+  subheader: string;
+  subheaderFontSize: number;
+  showTrendLine: boolean;
+  startYAxisAtZero: boolean;
+  timeRangeFixed: boolean;
+  trendLineData?: TimeSeriesDatum[];
+  mainColor: string;
+};
+
+class BigNumberVis extends React.PureComponent<BigNumberVisProps, {}> {
+  private gradientId: string = shortid.generate();
+
+  static defaultProps = {
+    bigNumberFallback: null,
```

Comment:
Do not need to specify `null` as default props and use default behavior `undefined`. 
`null` will also do not match the type `bigNumberFallback?: TimeSeriesDatum;` which is `TimeSeriesDatum | undefined`.

Same with `fromDatetime`, `toDatetime`, and `trendLineData`